### ISSUE TITLE
net.box: call lua func with raw msgpack

### ISSUE THREAD

**Issue by Khatskevich** [tarantool/tarantool#3349]
In some cases stored Lua procedure require raw msgpack (see below), however, net box converts all msgpacks to lua tables which makes the procedure to convert it back to msgpack, which is very CPU intensive.

There are two options of how to implement this feature:
1. Implement hint on client (require extra option passed with each request)
2. Implement hint on server (require some crutches)

#### Avro schema
Here are results of perf measurements of our Avro-schema:
https://github.com/tarantool/avro-schema/wiki/Small-performance-bench
In short, if you pass to Avro Lua table instead of msgpack it works **8 times slower**!
That makes Avro jit super-duper features completely useless!

#### Vshard
As @Gerold103  said, there are cases in `vshard` where data is just transferred through a node without any processing, so, `encode` and `decode` are redundant.

**Comment by kyukhin** [comment]
Could you please propose an API for the feature?

**Comment by Gerold103** [comment]
I can propose an API for server-side variant of the feature. If the function must take raw MessagePack, a user specifies `raw_args = true` in `box.schema.func.create`. Then the function must be ready to get the raw MessagePack as cdata in a first argument and its size in the second one.

**Comment by Totktonada** [comment]
It worth to also allow to return raw msgpack from a function.

**Comment by karlovnv** [comment]
> It worth to also allow to return raw msgpack from a function.

@Totktonada, could you please create an issue?

**Comment by Totktonada** [comment]
> > It worth to also allow to return raw msgpack from a function.
> 
> @Totktonada, could you please create an issue?

Filed #4861.

See also #4641 re C functions.
